src/CabanaPD_Particles.hpp

```diff
@@ -225,63 +250,39 @@ class Particles<MemorySpace, PMB, Dimension>
         auto vol = sliceVolume();
         auto nofail = sliceNoFail();
 
-        auto created = Kokkos::View<bool*, memory_space>(
-            Kokkos::ViewAllocateWithoutInitializing( "particle_created" ),
-            num_particles );
-
         // Initialize particles.
-        int mpi_rank = -1;
-        MPI_Comm_rank( MPI_COMM_WORLD, &mpi_rank );
-        int local_num_create = 0;
-        Kokkos::parallel_reduce(
-            "CabanaPD::Particles::init_particles_uniform",
-            Cajita::createExecutionPolicy( owned_cells, exec_space ),
-            KOKKOS_LAMBDA( const int i, const int j, const int k,
-                           int& create_count ) {
-                // Compute the owned local cell id.
-                int i_own = i - owned_cells.min( Cajita::Dim::I );
-                int j_own = j - owned_cells.min( Cajita::Dim::J );
-                int k_own = k - owned_cells.min( Cajita::Dim::K );
-                int pid = i_own + owned_cells.extent( Cajita::Dim::I ) *
-                                      ( j_own + k_own * owned_cells.extent(
-                                                            Cajita::Dim::J ) );
-
-                // Get the coordinates of the cell.
-                int node[3] = { i, j, k };
-                double cell_coord[3];
-                local_mesh.coordinates( Cajita::Cell(), node, cell_coord );
-
-                // Set the particle position.
-                for ( int d = 0; d < 3; d++ )
-                {
-                    x( pid, d ) = cell_coord[d];
-                    u( pid, d ) = 0.0;
-                    y( pid, d ) = 0.0;
-                    v( pid, d ) = 0.0;
-                    f( pid, d ) = 0.0;
-                }
-                // FIXME: hardcoded.
-                type( pid ) = 0;
-                nofail( pid ) = 0;
-                rho( pid ) = 1.0;
-
-                // Get the volume of the cell.
-                int empty[3];
-                vol( pid ) = local_mesh.measure( Cajita::Cell(), empty );
-
-                // Customize new particle.
-                // created( pid ) = create_functor( px, particle );
-                created( pid ) = true;
-
-                // If we created a new particle insert it into the
-                // list.
-                if ( created( pid ) )
-                {
-                    ++create_count;
-                }
-            },
-            local_num_create );
-        n_local = local_num_create;
+        auto create_functor =
+            KOKKOS_LAMBDA( const int pid, const double px[dim], const double pv,
+                           typename plist_x_type::particle_type& particle )
```

Comment:
@streeve : I assume "px[dim]" are particle reference positions? 

What is "pv"? Is that particle initial velocities? Why don't we have "pv[dim]" as for px?

Reply:
@streeve : I just noted below that "pv" is the particle volume.